promotion/datastore.go

```diff
@@ -628,6 +633,73 @@ func (pg *Postgres) SaveClaimCreds(creds *ClaimCreds) error {
 	return err
 }
 
+// GetDrainPoll Get the status of the drain poll job
+func (pg *Postgres) GetDrainPoll(drainID *uuid.UUID) (*DrainPoll, error) {
+	type dbDrainPoll struct {
+		ID         *uuid.UUID `db:"batch_id"`
+		Completed  bool       `db:"completed"`
+		Pending    bool       `db:"pending"`
+		Delayed    bool       `db:"delayed"`
+		InProgress bool       `db:"inprogress"`
+	}
+	var (
+		drainPoll = new(dbDrainPoll)
+		err       error
+	)
+
+	statement := `
+select
+	batch_id,
+	bool_and(completed) as completed,
+	bool_or(erred) as delayed,
+	(not bool_and(completed) and not bool_or(erred)) as inprogress,
+	(not bool_or(completed)) as pending
```

Comment:
would this line cause the initial state to be `unknown`?

Reply:
I think unknown state is impossible to achieve, as if there is a record, either completed/delayed/inprogress/pending would return a true value.  The unknown fail-safe should not be reached imo.

Reply:
tests showing how this unknown state are achieved have been added.  https://github.com/brave-intl/bat-go/pull/654/commits/d1d99642493fd663e8d5971aea7527d6171befb0